File: Anul 2/Semester 1/MAP/Seminar9/Seminar9/domain/MessageTask.cs
using System;

namespace Seminar9.domain
{
    public class MessageTask:Task
    {
        private Message msg;
        public DateTime data;

        public MessageTask():base(0,"") { }

        public MessageTask(int id, string desc, Message msg, DateTime data):base(id,desc)
        {
            this.msg = msg;
            this.data = data;
        }

        public Message Msg
        {
            get => msg;
            set => msg = value;
        }

        public MessageTask(DateTime data)
        {
            this.data = data;
        }

        public override void Execute()
        {
            Console.WriteLine(base.ToString() + " " + Msg.ToString() + " " + data.ToString());
        }
    }
}

File: Anul 2/Semester 1/MAP/Seminar 1/src/com/company/sem1/model/MessageTask.java
package com.company.sem1.model;

import com.company.sem1.Constants;

import java.time.LocalDateTime;

public class MessageTask extends Task {
    private String message;
    private String from;
    private String to;
    private LocalDateTime date;

    public MessageTask(String taskId, String desc, String message, String from, String to, LocalDateTime date) {
        super(taskId, desc);
        this.message = message;
        this.from = from;
        this.to = to;
        this.date = date;
    }

    public void execute() {
        System.out.println(toString());

    }

    @Override
    public String toString() {
        return super.toString()+ " " + message + " " + from + "  " + to + " " + " " + date.format(Constants.DATE_TIME_FORMATTER);

    }
}
